### PR TITLE
Executor imports interface, via code generation

### PR DESCRIPTION
**Pull request by andrei-marinica**
The initial code generator.

Much has changed in part 2, so do not be too critical. It has no direct influence on the code as it stands currently.

### CODE REVIEW

**Review by ovstinga — DISMISSED** [review]

**Review by sasurobert — COMMENTED** [review]
Why do we need this ?

File: arwen/elrondapi/generateMain/generateApi.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	return &eapigen.EIMetadata{
+		Groups: []*eapigen.EIGroup{
+			&eapigen.EIGroup{SourcePath: "bigFloatOps.go"},
+			&eapigen.EIGroup{SourcePath: "bigIntOps.go"},
+			&eapigen.EIGroup{SourcePath: "elrondei.go"},
+			&eapigen.EIGroup{SourcePath: "generateOps.go"},
+			&eapigen.EIGroup{SourcePath: "managedei.go"},
+			&eapigen.EIGroup{SourcePath: "manBufOps.go"},
+			&eapigen.EIGroup{SourcePath: "smallIntOps.go"},
+			&eapigen.EIGroup{SourcePath: "../cryptoapi/cryptoei.go"},
```
Comment (camilbancioiu) [inline arwen/elrondapi/generateMain/generateApi.go:22]:
```suggestion
			{SourcePath: "bigFloatOps.go"},
			{SourcePath: "bigIntOps.go"},
			{SourcePath: "elrondei.go"},
			{SourcePath: "generateOps.go"},
			{SourcePath: "managedei.go"},
			{SourcePath: "manBufOps.go"},
			{SourcePath: "smallIntOps.go"},
			{SourcePath: "../cryptoapi/cryptoei.go"},
```
The struct literals are implicit here.

**Comment by ovstinga** [comment]
Writing 240+ interfaces by hand is more likely to be error prone.
Generating them with this tool, we are more confident we don't miss anything and marking them as autogenerated will be better to guard against human error in the future.

**Review by camilbancioiu — DISMISSED** [review]

File: arwen/elrondapi/generate/eiGenModel.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+// EIFunctionArg models the executor callback method result.
+type EIFunctionResult struct {
+	Type string
+}
+
+// EIFunction holds data about one function in the VM EI.
+type EIFunction struct {
+	OriginalName    string
+	LowerCaseName   string
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply arwen/elrondapi/generate/eiGenModel.go:17]: it isn't the wasm import name, it doesn't have the prefix
it is just the name in lowecase

File: arwen/elrondapi/generate/eiGenModel.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	CapitalizedName string
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply arwen/elrondapi/generate/eiGenModel.go:18]: It doesn't have to be the CallbackName only, the writer decides this

File: arwen/elrondapi/generate/eiGenParse.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+			}
+		}
+	}
+	return arguments, nil
+}
+
+func extractEIFunctionResult(decl *ast.FuncDecl) (*EIFunctionResult, error) {
+	if decl.Type.Results == nil {
+		return nil, nil
+	}
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply arwen/elrondapi/generate/eiGenParse.go:44]: I thought so too until it crashed

**Pull request merged**